pkg/controller/infinispan/autoscale.go

```diff
@@ -0,0 +1,204 @@
+package infinispan
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	infinispanv1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	"github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	consts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var autoscaleThreadPool = struct {
+	sync.RWMutex
+	m map[types.NamespacedName]int
+}{m: make(map[types.NamespacedName]int)}
+
+// Starting a go routine that does polling autoscaling on an Infinispan cluster
+func addAutoscalingEquipment(clusterNsn types.NamespacedName, r *ReconcileInfinispan) {
+	autoscaleThreadPool.Lock()
```

Comment:
@rigazilla looks like this is good implementation and avoid conflicts between separate clusters